File: TrelloApi/TrelloApi/src/Domain/Entities/CardLabel.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TrelloApi.Domain.Entities;

[Table("CardLabel")]
public class CardLabel: Entity
{
    [ForeignKey("Card")]
    public int CardId { get; set; }
    public Card Card { get; set; }
        
    [ForeignKey("Label")]
    public int LabelId { get; set; }
    public Label Label { get; set; }

    public CardLabel(int cardId, int labelId)
    {
        CardId = cardId;
        LabelId = labelId;
        CreatedAt = DateTime.UtcNow;
        UpdatedAt = null;
    }
}

File: TrelloApp/src/app/core/models/card-label.ts
import {Card} from './card';
import {Label} from './label';

export interface CardLabel {
  id: number,
  createdAt: string,
  updatedAt?: string,
  cardId: number,
  labelId: number,
  card: Card,
  label: Label
}

export interface AddCardLabel {
  labelId: number,
}
